core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -973,22 +1027,32 @@ class TimeSeriesShard(val dataset: Dataset,
         s"shard=$shardNum", e); disableAddPartitions()
     }
 
+  /**
+    * Creates new partition and adds them to the shard data structures. DOES NOT update
+    * lucene index. It is the caller's responsibility to add or skip that step depending on the situation.
+    */
   protected def createNewPartition(partKeyBase: Array[Byte], partKeyOffset: Long,
-                                   group: Int, initMapSize: Int = initInfoMapSize): TimeSeriesPartition =
+                                   group: Int, usePartId: Option[Int],
```

Comment:
Instead of an Option[Int], can we pass two params, an Int and a reuseFlag, to avoid allocation of Option?

Reply:
will fix